sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -145,6 +145,20 @@ private[hive] class HiveClientImpl(
         warehouseDir.foreach { dir =>
           ret.getConf.setVar(ConfVars.METASTOREWAREHOUSE, dir)
         }
+        // Since in jdk11, when HiveClient's withHiveSate, it will set ThreadLocal SessionState,
+        // In HiveThriftServer2, it will call HiveUtils.newClientForExecution() to get a client
+        // for Execution, then that method will trigger here to execute, in that case we can't reset
+        // ret.getConf's ClassLoader.
+        if (HiveUtils.isCliSessionState) {
```

Comment:
@srowen @wangyum  

Some special point for jdk11.  Need to add more explanation？